tests/test_glm.py

```diff
@@ -1697,17 +1758,308 @@ def test_reg_set_params(self, params, warns):
         "params, warns",
         [
             # set regularizer str only
-            ({"regularizer_strength": 1.}, does_not_raise()),
-            ({"regularizer_strength": None},
-             pytest.warns(UserWarning, match="Caution: regularizer strength has not been set")),
-        ]
+            ({"regularizer_strength": 1.0}, does_not_raise()),
+            (
+                {"regularizer_strength": None},
+                pytest.warns(
+                    UserWarning, match="Caution: regularizer strength has not been set"
+                ),
+            ),
+        ],
     )
     @pytest.mark.parametrize("reg", ["Ridge", "Lasso", "GroupLasso"])
     def test_reg_set_params_reg_str_only(self, params, warns, reg):
         model = nmo.glm.GLM(regularizer=reg, regularizer_strength=1)
         with warns:
             model.set_params(**params)
 
+    @pytest.mark.parametrize(
+        "solver_name, reg",
+        [
+            ("SVRG", "Ridge"),
+            ("SVRG", "UnRegularized"),
+            ("ProxSVRG", "Ridge"),
+            ("ProxSVRG", "UnRegularized"),
+            ("ProxSVRG", "Lasso"),
+            ("ProxSVRG", "GroupLasso"),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "obs",
+        [
+            nmo.observation_models.PoissonObservations(
+                inverse_link_function=jax.nn.softplus
+            )
+        ],
+    )
+    @pytest.mark.parametrize("batch_size", [None, 1, 10])
+    @pytest.mark.parametrize("stepsize", [None, 0.01])
+    def test_glm_optimal_config_set_initial_state(
+        self,
+        solver_name,
+        batch_size,
+        stepsize,
+        reg,
+        obs,
+        poissonGLM_model_instantiation,
+    ):
+        X, y, _, true_params, _ = poissonGLM_model_instantiation
+        if reg == "GroupLasso":
+            reg = nmo.regularizer.GroupLasso(mask=jnp.ones((1, X.shape[1])))
+        model = nmo.glm.GLM(
+            solver_name=solver_name,
+            solver_kwargs=dict(batch_size=batch_size, stepsize=stepsize),
+            observation_model=obs,
+            regularizer=reg,
+            regularizer_strength=None if reg == "UnRegularized" else 1.0,
+        )
+        opt_state = model.initialize_state(X, y, true_params)
+        solver = inspect.getclosurevars(model._solver_run).nonlocals["solver"]
+
+        if stepsize is not None:
+            assert opt_state.stepsize == stepsize
+            assert solver.stepsize == stepsize
+        else:
+            assert opt_state.stepsize > 0
+            assert isinstance(opt_state.stepsize, float)
+            model.fit(X, y)
```

Comment:
I would consider removing `model.fit()` calls in this test (as well as in `test_glm_optimal_config_set_initial_state_pytree`, and both functions with the same names in the `TestPopulationGLM` class) since its not being consistently called across all cases, and since no checks are happening after the model is fit

Reply:
yes, I'm confused about this `model.fit` too -- is it supposed to be in the else block?